vb-fe/src/layout/Layout.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+
+import { Box, Flex } from "@chakra-ui/react";
+
+import { MIN_SHEET_HEIGHT } from "util/Variables";
+
+interface GenericLayoutProps {
+  children?: ReactElement | ReactElement[];
+  className?: string;
+  flex?: string;
+  show?: boolean;
+}
+
+interface LayoutWrapperProps extends Omit<GenericLayoutProps, "children"> {
+  children?:
+    | ReactElement<typeof LayoutHeader | typeof LayoutBody | typeof LayoutFooterOverlay | typeof LayoutFooter>
+    | Array<ReactElement<typeof LayoutHeader | typeof LayoutBody | typeof LayoutFooterOverlay | typeof LayoutFooter>>;
+}
+
+interface LayoutBodyProps extends Omit<GenericLayoutProps, "children"> {
+  children?:
+    | ReactElement<typeof LayoutSidebar | typeof LayoutContent>
+    | Array<ReactElement<typeof LayoutSidebar | typeof LayoutContent>>;
+}
+
+export const LayoutFooterOverlay: FunctionComponent<GenericLayoutProps> = (): JSX.Element => {
+  useEffect(() => {
+    document.body.style.overflow = "hidden";
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      document.body.style.overflow = "visible";
+    };
+  });
+
+  return (
+    <Box
+      h="100vh"
+      w="100vw"
+      bgColor="blackAlpha.700"
+      position="fixed"
+      top="0"
+      left="0"
+      zIndex="998"
+      overflowY="hidden"
+    />
+  );
+};
+
+export const LayoutFooter: FunctionComponent<GenericLayoutProps> = ({ show = true, children, className }) => {
```

Comment:
This is pretty cool.  Something you might be interested in is the ability to do something like

```
export const Layout = () => { ... }

Layout.Footer = () => { ... }

Layout.Content = () => { ... }
```

It's definitely not necessary, but I'm not sure if you've ever seen this before, and I think it's pretty cool lol

Reply:
yusss, i've seen this I can do this! in this PR or another one?

Reply:
i'll just do it in this one